### DISCUSSION TITLE
Daydreaming about SSH

### DISCUSSION THREAD

**Discussion by ghost** [micropython/micropython#9422] [General]
I noticed that [libssh2](https://www.libssh2.org/) supports mbedTLS...

Is there a way, if only theoretical, to run a SSH client on MicroPython? Maybe with C and natmods?

**Comment by jimmo** [comment]
Definitely possible. It has been done in the past -- see for example https://github.com/loboris/MicroPython_ESP32_psRAM_LoBo/wiki/ssh  (this is an old fork, that I believe is now unmaintained).

**Comment by cwalther** [comment]
Here’s an implementation of an SSH server (in Python, even), maybe it could be extended into a client (I have no idea how symmetric SSH is): https://github.com/LarsH/larssh (from [the old forum](https://forum.micropython.org/viewtopic.php?t=2030&p=69143#p69143)).
